### PR TITLE
pullToRefresh adjustments - fixing scrolling hijacking

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Bug Fix
- [x] Optimization

## Description

We've been noticing some issues with horizontally scrollable elements on touch devices, like the most recent one: editor toolbar. It was just a little "picky" - sometimes working and sometimes not working. I've built a little [prototype in codepen](https://codepen.io/ludwiczakpawel/full/eYGmxGj) to validate whether it's me going crazy or is this really something related to our codebase...

Anyway, what initially felt like some [weird CSS bug](https://github.com/forem/forem/pull/15615), turned out to be caused by our pullToRefresh library which is hijacking some touch events (that's why it's been happening only on touch devices). This library is primarily useful only on Forem iOS app but not really in actual browsers (Safari, Chrome), since all the browsers support "pull to refresh" behavior natively. 

As a short-term solution, we've decided to disable this library on `/new` view (only) since it's pointless there anyway. This fixed the issue with editor toolbar but it's still a problem elsewhere (for example, we have scrollable tabs on the Search Results view, on mobile). As an addition I've also slightly adjusted this library settings making this behavior a tiny bit better (but **not** eliminating the problem entirely). And that's what this PR does.

As for long-term solution, I think we should (cc @fdoxyz):

1. Disable pullToRefresh library on actual web view but keep it enabled for Forem iOS app if needed.
2. Drop this library entirely and rely on some native solution for Forem iOS app.

## Related Tickets & Documents

- [This slack thread](https://forem-team.slack.com/archives/C7GE84DEJ/p1638390238247900).
- [Related PR](https://github.com/forem/forem/pull/15615) that I thought gonna fix the issue.

## QA Instructions, Screenshots, Recordings

1. Here's how to review PRs involving actual mobile devices: https://forem.team/fdoxyz/how-to-review-prs-with-the-mobile-experience-in-mind-13mg (I highly recommend ngrok solution).
2. Once it's done, just navigate to /new and see if scrolling behavior on editor toolbar performs slightly better than it is right now on production `dev.to/new`. Also feel free to test whether pullToRefresh works as expected on other views (since it will be disabled on `/new`)

### UI accessibility concerns?

If anything, this improves the accessibility.

## Added/updated tests?

- [x] No need. 

## [Forem core team only] How will this change be communicated?

- [x] No need.

### CODE REVIEW

File: app/javascript/crayons/MarkdownToolbar/MarkdownToolbar.jsx [hunk 1]
```diff
@@ -291,8 +291,8 @@ export const MarkdownToolbar = ({ textAreaId }) => {
           icon={icon}
           className={
             isOverflow
-              ? 'overflow-menu-btn hidden m:block mr-2'
-              : 'toolbar-btn m:hidden mr-2'
+              ? 'overflow-menu-btn hidden m:block mr-1'
```
Comment (ludwiczakpawel) [inline app/javascript/crayons/MarkdownToolbar/MarkdownToolbar.jsx:294]: Sneaking-in little visual fix for toolbar :P

File: app/assets/stylesheets/views/article-form.scss [hunk 2]
```diff
@@ -167,22 +167,23 @@
     position: sticky;
     top: 0;
     background: var(--base-0);
-    padding: 0 var(--content-padding-x);
+    padding: var(--su-2) var(--content-padding-x);
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/views/article-form.scss:170]: This basically reverts some of the previous changes that were meant to address scrolling issue...

File: app/javascript/article-form/components/Toolbar.jsx [hunk 3]
```diff
@@ -10,13 +10,11 @@ export const Toolbar = ({ version, textAreaId }) => {
...
-          <ImageUploader editorVersion={version} />
-        ) : (
-          <MarkdownToolbar textAreaId={textAreaId} />
-        )}
-      </div>
+      {version === 'v1' ? (
+        <ImageUploader editorVersion={version} />
+      ) : (
+        <MarkdownToolbar textAreaId={textAreaId} />
+      )}
```
Comment (ludwiczakpawel) [inline app/javascript/article-form/components/Toolbar.jsx:17]: This reverts some of the previous changes that were meant to address scrolling issue...

File: app/assets/javascripts/lib/pulltorefresh.js [hunk 4]
```diff
@@ -53,7 +53,8 @@
           document.getElementById('article-body') ||
           document.getElementById('listings-index-container')) &&
         !document.body.classList.contains('modal-open') &&
-        document.body.dataset.leftNavState !== 'open'
+        document.body.dataset.leftNavState !== 'open' &&
+        !document.getElementById('article-form')
```
Comment (ludwiczakpawel) [inline app/assets/javascripts/lib/pulltorefresh.js:57]: This disables PTR on `/new`

File: app/assets/javascripts/lib/pulltorefresh.js [hunk 5]
```diff
@@ -22,7 +22,7 @@
     distThreshold: 60,
     distMax: 80,
     distReload: 50,
-    distIgnore: 0,
+    distIgnore: 10,
```
Comment (ludwiczakpawel) [inline app/assets/javascripts/lib/pulltorefresh.js:25]: This slightly improves the general behavior of PTR making it a little bit "delayed".

**Review by fdoxyz — APPROVED** [review]
Gave it a try and works great! Definitely fixes the horizontal scroll issue in the editor toolbar 🙌 

> Drop this library entirely and rely on some native solution for Forem iOS app.

I wholeheartedly agree with this. A quick spike to make this work on mobile I think is very feasible in the short term. I'll reach out to you and others to test it out hopefully soon. Getting rid of it altogether would be for the best IMHO

**Review by aitchiss — APPROVED** [review]

**Review by joshpuetz — COMMENTED** [review]
Tested locally on iOS 15/Forem-ios latest and working!

**Pull request merged**